src/CLI/Config.php

```diff
@@ -17,15 +16,37 @@ public function __construct()
         $this->classSetRules = [];
     }
 
-    public function add(ClassSet $classSet, ArchRule ...$rules): self
+    public function add(ClassSet $classSet, array $rules): self
     {
-        $this->classSetRules[] = ClassSetRules::create($classSet, ...$rules);
+        $this->classSetRules[] = ClassSetRules::create($classSet, $rules);
 
         return $this;
     }
 
-    public function getClassSetRules(): array
+    public function getClassSetRules(?string $ruleFilter = null): array
     {
+        if (null !== $ruleFilter) {
+            return $this->filterRuleIntoClassSetRules($ruleFilter);
+        }
+
+        return $this->classSetRules;
+    }
+
+    private function filterRuleIntoClassSetRules(string $ruleFilter): array
+    {
+        /** @var ClassSetRules $classSetRules */
+        foreach ($this->classSetRules as $index => $classSetRules) {
```

Comment:
why don't we create a collection of classSetRules instead of using a plain array?